common/src/main/scala/explore/utils/package.scala

```diff
@@ -3,9 +3,23 @@
 
 package explore
 
+import cats.data.State
+import monocle.Lens
+import monocle.state.all._
+
 package object utils {
 
   def abbreviate(s: String, maxLength: Int): String =
     if (s.length > maxLength) s"${s.substring(0, maxLength)}\u2026" else s
 
+  implicit final class LensEditorOps[S, A](self: Lens[S, A]) {
+    def edit(a: A): State[S, A]         =
+      self.assign(a)
+    def :=(a:   A): State[S, A]         =
+      edit(a)
+    def edit(a: Option[A]): State[S, A] =
+      a.fold(self.st)(self.assign)
+    def :=(a:   Option[A]): State[S, A] =
+      edit(a)
+  }
```

Comment:
With the non-`Option` versions included, maybe it would make sense to share these in `core` syntax.

Reply:
Also, maybe we want the methods `@inline`, at least the aliases (`:=`).

Reply:
BTW, this was copied from @swalker implementation. I just added an extra or non `Option`. We should move it to core.